synapse/handlers/profile.py

```diff
@@ -38,11 +43,79 @@ def __init__(self, hs):
 
         self.user_directory_handler = hs.get_user_directory_handler()
 
+        self.http_client = hs.get_simple_http_client()
+
         if hs.config.worker_app is None:
             self.clock.looping_call(
                 self._update_remote_profile_cache, self.PROFILE_UPDATE_MS,
             )
 
+            if len(self.hs.config.replicate_user_profiles_to) > 0:
+                reactor.callWhenRunning(self._assign_profile_replication_batches)
+                reactor.callWhenRunning(self._replicate_profiles)
+                self.clock.looping_call(
```

Comment:
It'd be good to spell out in a comment why we're looping _replicate_profiles.  I can see we'll need to do a catchup job at launch, but why do we then keep running it every 2 minutes, given whenever we set profile data it kicks off a sync anyway?

Reply:
done